addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/dto/HueDevice.java

```diff
@@ -352,10 +352,35 @@ private void setState(State itemState) {
             successApplied.put("effect", newState.effect); // Pretend that worked
         }
         if (newState.xy != null) {
-            errorApplied.add("xy");
+            try {
+                HueStateColorBulb c = as(HueStateColorBulb.class);
+                c.colormode = ColorMode.xy;
+                c.bri = as(HueStateBulb.class).bri;
+                c.xy[0] = newState.xy.get(0);
+                c.xy[1] = newState.xy.get(1);
+                command = c.toHSBType();
+                successApplied.put("xy", newState.xy);
+            } catch (ClassCastException e) {
+                errorApplied.add("xy");
+            }
         }
         if (newState.xy_inc != null) {
-            errorApplied.add("xy_inc");
+            try {
+                HueStateColorBulb c = as(HueStateColorBulb.class);
+                double newX = c.xy[0] + newState.xy_inc.get(0);
+                double newY = c.xy[1] + newState.xy_inc.get(1);
+                if (newX < 0 || newX > 1 || newY < 0 || newY > 1) {
+                    throw new ClassCastException();
```

Comment:
Do not throw standard exception types such as the `ClassCastException `, use a specific exception and if none exists create your own.

Reply:
I would throw a IllegalArgumentException and catch it in the surrounding block next to the ClassCastException.

Reply:
Does that also apply to the lines above where the same things are done? Not part of this PR but we're in the same method here, there should be some consistency (in a different method or even different class I would be more relaxed about it, but in the same method just like 10 lines apart there has to be consistency)

Reply:
Yes, I was lazy while writing this method apparently. Would be awesome if you could fix the other instances, too.